server/src/test/java/org/elasticsearch/cluster/ClusterStateTests.java

```diff
@@ -293,25 +293,25 @@ public void testToXContentWithMultipleProjects() throws IOException {
                     },
                     "projects": [
                       {
-                        "id": "tb5W0bx765nDVIwqJPw92G",
+                        "id": "3LftaL7hgfXAsF60Gm6jcD",
```

Comment:
What happened here?

Reply:
It looks like our calculation creates a bigger maps in some cases. This results in a different iteration order in several tests.

Reply:
```
expected size=0 old_capacity=1 new_capacity=0 <--
expected size=1 old_capacity=2 new_capacity=2
expected size=2 old_capacity=3 new_capacity=3
expected size=3 old_capacity=5 new_capacity=4 <--
expected size=4 old_capacity=6 new_capacity=6
expected size=5 old_capacity=7 new_capacity=7
expected size=6 old_capacity=9 new_capacity=8 <--
expected size=7 old_capacity=10 new_capacity=10
expected size=8 old_capacity=11 new_capacity=11
expected size=9 old_capacity=13 new_capacity=12 <--
expected size=10 old_capacity=14 new_capacity=14
expected size=11 old_capacity=15 new_capacity=15
expected size=12 old_capacity=17 new_capacity=16 <--
```